apps/emqx_auth_jwt/src/emqx_authn_jwt.erl

```diff
@@ -219,14 +219,24 @@ verify(undefined, _, _, _) ->
 verify(JWT, JWKs, VerifyClaims, AclClaimName) ->
     case do_verify(JWT, JWKs, VerifyClaims) of
         {ok, Extra} ->
-            {ok, acl(Extra, AclClaimName)};
+            try
+                {ok, acl(Extra, AclClaimName)}
+            catch
+                throw:{bad_acl_rule, Reason} ->
+                    %% it's a invalid token, so ok to log
+                    ?TRACE_AUTHN_PROVIDER("bad_acl_rule", Reason#{jwt => JWT}),
```

Comment:
Previously, the validation was lazy, that is, invalid rules were handled on authz stage. This is kept for v1 rules;
For v2 we now break connection, don't we consider this to be a breaking change?

Reply:
A invalid token will cause any pub/sub action to fail, maybe it's better to close connection early.
It's a new format, so maybe it's not entirely a breaking change since all the v1 clients will not notice any change.